tests/mir-opt/pre-codegen/loops.rs

```diff
@@ -0,0 +1,54 @@
+// compile-flags: -O -Zmir-opt-level=2 -g
+// ignore-debug
+
+#![crate_type = "lib"]
+
+pub fn int_range(start: usize, end: usize) {
+    for i in start..end {
+        opaque(i)
+    }
+}
+
+pub fn vec_range(mut v: Vec<impl Sized>) {
+    for i in 0..v.len() {
+        let x = &mut v[i];
+        opaque((i, x))
+    }
+    for i in 0..v.len() {
+        let x = &v[i];
+        opaque((i, x))
+    }
+}
+
+pub fn vec_iter(mut v: Vec<impl Sized>) {
```

Comment:
This means that the dropping is included; is that intentional?  It seems like a distraction from the "loops" part that the filename suggests this cares about.

Also, vec's `iter` and `iter_mut` are the same as slice's, so I would suggest removing this part since we already have <https://github.com/rust-lang/rust/blob/master/tests/mir-opt/pre-codegen/slice_iter.rs#L27>.